CDCgov/prime-reportstream
# Implements table name into LookupTableApi calls

This PR following #5909 this PR will swap out the enumerated value of `SENDER_AUTOMATION_TABLE_ROW` with the true table name in all API calls to the lookuptable api.

Test Steps:
1. Log in as a PrimeAdmin
2. Ensure the Value Sets list loads -> select one and go to its detail page
3. Ensure the details page loads -> try to edit one and save
4. Ensure the save and activation calls in the network tab conclude with 200s
5. Check all network calls to this api (`content`, `activate`, and a call titled the same as your table should appear in your network tab upon save) use the new format of URL:

NEW
```
http://localhost:3000/api/lookuptables/gender/{version}/content
# where "gender" is the name of any lookup table
```
OLD
```
http://localhost:3000/api/lookuptables/sender_automation_table_row/{version}/content
# where "sender_automation_table_row" was an enumerated value
```

Known Issues:

When saving a new value, creating a new version, then swapping the value _back_ to its previous value, the following error occurs. Seems related to API functionality, not front-end calling of the API:
<img width="461" alt="Screen Shot 2022-08-25 at 3 15 06 PM" src="https://user-images.githubusercontent.com/4022304/186750467-934d98e4-2868-480f-b798-e63e081b5213.png">


## Changes
- Hook layer of LookupTableApi -> piping the true table name through instead of replacing with enumerated value

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #5947 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
